src/Build/Evaluation/SimpleProjectRootElementCache.cs

```diff
@@ -0,0 +1,127 @@
+    /// </summary>
+    internal class SimpleProjectRootElementCache : ProjectRootElementCacheBase
+    {
+        private readonly ConcurrentDictionary<string, ProjectRootElement> _cache;
+
+        internal SimpleProjectRootElementCache()
+        {
+            _cache = new ConcurrentDictionary<string, ProjectRootElement>(StringComparer.OrdinalIgnoreCase);
+            LoadProjectsReadOnly = true;
+        }
+
+        internal override ProjectRootElement Get(
+            string projectFile,
+            OpenProjectRootElement openProjectRootElement,
+            bool isExplicitlyLoaded,
+            bool? preserveFormatting)
+        {
+            // Should already have been canonicalized
+            ErrorUtilities.VerifyThrowInternalRooted(projectFile);
+
+            return openProjectRootElement == null
+                ? GetFromCache(projectFile)
+                : GetFromOrAddToCache(projectFile, openProjectRootElement);
+        }
+
+        private ProjectRootElement GetFromCache(string projectFile)
+        {
+            if (_cache.TryGetValue(projectFile, out ProjectRootElement projectRootElement))
+            {
+                return projectRootElement;
+            }
+
+            return null;
+        }
+
+        private ProjectRootElement GetFromOrAddToCache(string projectFile, OpenProjectRootElement openFunc)
+        {
+            return _cache.GetOrAdd(projectFile, key =>
+            {
+                var rootElement = openFunc(key, this);
+                ErrorUtilities.VerifyThrowInternalNull(rootElement, "projectRootElement");
+                ErrorUtilities.VerifyThrow(rootElement.FullPath.Equals(key, StringComparison.OrdinalIgnoreCase),
+                    "Got project back with incorrect path");
+                ErrorUtilities.VerifyThrow(_cache.TryGetValue(key, out _),
+                    "Open should have renamed into cache and boosted");
+
+                return rootElement;
+            });
+        }
+
+        internal override void AddEntry(ProjectRootElement projectRootElement)
+        {
+            if (_cache.TryAdd(projectRootElement.FullPath, projectRootElement))
+            {
+                RaiseProjectRootElementAddedToCacheEvent(projectRootElement);
+            }
+        }
+
+
+        internal override void RenameEntry(string oldFullPath, ProjectRootElement projectRootElement)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal override ProjectRootElement TryGet(string projectFile)
+        {
+            return TryGet(projectFile, null);
+        }
+
+
+        internal override ProjectRootElement TryGet(string projectFile, bool? preserveFormatting)
+        {
+            return Get(
+                projectFile,
+                null,
+                false,
+                preserveFormatting);
+        }
+
+        internal override void DiscardStrongReferences()
```

Comment:
`throw new NotImplementedException();`?

Reply:
Discussed w/ Mihai offline, found it easier to just no-op here since 
```ProjectCollection.UnloadProject()``` calls this, and we preferred this over doing a type check on the simple cache.